airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionTranslator.kt

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+package io.airbyte.cdk.integrations.util
+
+import io.airbyte.cdk.integrations.base.errors.messages.ErrorMessage
+import io.airbyte.commons.exceptions.ConfigErrorException
+import io.airbyte.commons.exceptions.ConnectionErrorException
+import io.airbyte.commons.exceptions.TransientErrorException
+import io.github.oshai.kotlinlogging.KotlinLogging
+
+private val LOGGER = KotlinLogging.logger {}
+
+const val COMMON_EXCEPTION_MESSAGE_TEMPLATE: String =
+    "Could not connect with provided configuration. Error: %s"
+
+const val DATABASE_CONNECTION_ERROR: String =
+    "Encountered an error while connecting to the database error"
+
+class ConnectorErrorProfile {
+    var id: Int = 0
+    var errorClass: String? = null
+    var regexMatchingPattern: String? = null
+    var errorType: String? = null // config, transient, system
+    var externalMessage: String? = null
+    var sampleInternalMessage: String? = null
+    var githubIssues: List<String>? = null
```

Comment:
Maybe this should be `referenceLink` which can be a github issue, a zendesk ticket, etc.

Reply:
agreed.

Reply:
fixed.